src/Plugin/yaml_content/process/ProcessFile.php

```diff
@@ -21,21 +22,21 @@ class ProcessFile extends File {
    * {@inheritdoc}
    */
   public function process(ProcessingContext $context, array &$field_data) {
-    $entity_type = $this->configuration[0];
-    $filter_params = $this->configuration[1];
+    [$entity_type, $filter_params] = $this->configuration;
 
     $filename = $filter_params['filename'];
     $directory = '/data_files/';
     // If the entity type is an image, look in to the /images directory.
-    if ($entity_type == 'image') {
+    if ($entity_type === 'image') {
       $directory = '/images/';
     }
     // Path set tide_demo_content directory.
     $output = file_get_contents($context->getContentLoader()->getContentPath() . $directory . $filename);
     if ($output !== FALSE) {
       $destination = 'public://';
-      // Look-up the field's directory configuation.
-      if ($directory = $context->getField()->getSetting('file_directory')) {
+      // Look-up the field's directory configuration.
+      $directory = $context->getField()->getSetting('file_directory');
+      if ($directory) {
         $directory = trim($directory, '/');
         $directory = PlainTextOutput::renderFromHtml('tide_demo_content');
         if ($directory) {
```

Comment:
@MdNadimHossain you may want to check these lines as they don't make sense to me. Not sure what was your intention here hence I couldn't fix the logic.